nixos/modules/hardware/video/nvidia.nix

```diff
@@ -285,7 +285,7 @@ in
             KERNEL=="nvidia_uvm", RUN+="${pkgs.runtimeShell} -c 'mknod -m 666 /dev/nvidia-uvm-tools c $$(grep nvidia-uvm /proc/devices | cut -d \  -f 1) 1'"
           '';
           hardware.opengl = {
-            extraPackages = [ nvidia_x11.out ];
+            extraPackages = [ nvidia_x11.out nvidia_x11.settings.libXNVCtrl ];
```

Comment:
:exclamation:  This will break things. Depending on config ( `useSettings` or `nvidiaSettings` ) `nvidia_x11.settings` might be a empty set.  See https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/os-specific/linux/nvidia-x11/generic.nix#L215
This should handle that case or be reverted.

Reply:
Submitted https://github.com/NixOS/nixpkgs/pull/316516 to address this.

Reply:
@megheaiulian thanks for the heads up!